### PR TITLE
[Merged by Bors] - Improve the `set_active_camera` system

### LINKED ISSUE #4227: Panic after despawning camera
bevy version: main 026563cb1e89de30af5b801c89e8fb51a83e4970

When spawning a camera and despawning it later it panics with
```
thread 'main' panicked at 'view entity should exist: QueryDoesNotMatch', [...]\crates\bevy_core_pipeline\src\main_pass_2d.rs:45:14
```

```rust
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup)
        .add_system(remove_camera)
        .run();
}

fn setup(mut commands: Commands) {
    commands.spawn_bundle(OrthographicCameraBundle::new_2d());
}

fn remove_camera(
    mut commands: Commands,
    query: Query<Entity, With<Camera>>,
    keyboard_input: Res<Input<KeyCode>>,
) {
    if keyboard_input.just_pressed(KeyCode::Space) {
        for entity in query.iter() {
            commands.entity(entity).despawn();
        }
    }
}
```

### PR DESCRIPTION
**Pull request by kirusfg**
# Objective

- Make `set_active_camera` system correctly respond to camera deletion, while preserving its correct behavior on first ever frame and any consequent frame, and with multiple cameras of the same type available in the world.
- Fixes #4227

## Solution

- Add a check that the entity referred to by `ActiveCamera` still exists in the world.

### CODE REVIEW

File: crates/bevy_render/src/camera/camera.rs [hunk 1]
```diff
@@ -270,12 +270,20 @@ pub fn set_active_camera<T: Component>(
...
-        return;
+    // Check if there is already an active camera set and
+    // that it has not been deleted on the previous frame
+    if let Some(camera) = active_camera.get() {
+        if cameras.contains(camera) {
+            return;
+        }
     }
 
+    // If the previous active camera ceised to exist -
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/camera/camera.rs:281]:
```suggestion
    // If the previous active camera ceased to exist
```

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
IMO we should tighten the query in the system you tweaked to add a With filter for Camera. It'll catch a few really weird edge cases when using the fallback you added.

**Comment by kirusfg** [comment]
Yep, you could definitely slap a `Camera3d` or `Camera2d` on anything and it would cause problems, but I hope Bevy users won't do that :)

I've included your suggestions, and I hope correctly.

**Review by alice-i-cecile — APPROVED** [review]
Perfect, that was exactly what I had in mind.

**Review by mockersf — APPROVED** [review]

**Review by superdump — APPROVED** [review]

**Pull request closed without merging**